### ISSUE TITLE
Update multiple container images at once similar to `kubectl patch` command

### ISSUE THREAD

**Issue by ashwiniramesha** [fabric8io/kubernetes-client#2247]
Hello,

Since rolling() API [doesn't work yet](https://github.com/fabric8io/kubernetes-client/issues/1868), I have the following calls to edit() to perform rolling updates when there is a change in the image versions. My deployment has multiple containers and I'm looking for a way to update them all in one rollign update and not do one rolling update per image. Is this something fabri8 API supports today? Any reference to documents would be helpful.

kubernetes-client version: 4.10.1

```
 for(Container container : containers) {
          String name = container.getName();
          String imgName = container.getImage();
          int index = containers.indexOf(container);
          DeploymentAfterPatch = getK8sClient().apps().deployments().inNamespace(namespace)
          .withName(deploymentName).edit().editSpec().editTemplate().editSpec()
          .editContainer(index).withName(name).withNewImage(imgName)
          .endContainer().endSpec().endTemplate().endSpec().done(); ;
        }

```

When I use `kubectl rollout history deployment <deploymentName>`, this shows one entry for each container.

**Comment by manusa** [comment]
Relates to: #1868

**Comment by rohanKanojia** [comment]
Once https://github.com/fabric8io/kubernetes-client/pull/2272 gets merged, you would be able to update multiple container images like this:

```
Map<String, String> containerToImageMap = new HashMap<>();
containerToImageMap.put("nginx", "nginx:perl");
containerToImageMap.put("sidecar", "someImage:someVersion");
Deployment updatedDeployment = client.apps().deployments()
      .inNamespace("default")
      .withName("nginx-deployment")
      .rolling()
      .updateImage(containerToImageMap);
```

**Comment by rohanKanojia** [comment]
@ashwiniramesha : HI, we released 4.10.3 yesterday. I just tested with a multicontainer deployment. I was able to update images like this:

```
try (KubernetesClient client = new DefaultKubernetesClient()) {
    String namespace = "default";
    Map<String, String> updatedImages = new HashMap<>();
    updatedImages.put("busybox", "busybox:1");
    updatedImages.put("hello-kubernetes", "paulbouwer/hello-kubernetes:1.8");

    client.apps().deployments().inNamespace(namespace).withName("hello-kubernetes")
            .rolling().updateImage(updatedImages);
}
```

**Comment by rohanKanojia** [comment]
I'm closing this issue for now, but feel free to reopen it in case you face any problems with it.
